io-engine/src/bdev/device.rs

```diff
@@ -504,6 +510,40 @@ impl BlockDeviceHandle for SpdkBlockDeviceHandle {
             source: Errno::ENXIO,
         })
     }
+    /// Flush the io in buffer to disk, for the Local Block Device.
+    async fn flush_io(&self) -> Result<u64, CoreError> {
+        let is_write_cache =
+            unsafe { spdk_bdev_has_write_cache(self.device.0.as_inner_ref()) };
+        if !is_write_cache {
+            info!("No Write Cache, No need of Flush");
+            return Ok(0);
+        }
+        let (desc, chan) = self.handle.io_tuple();
+        let bdev_size = self.device.size_in_bytes();
+        let (s, r) = oneshot::channel::<bool>();
+        let result = unsafe {
+            spdk_bdev_flush(
+                desc,
+                chan,
+                0,
+                bdev_size,
+                Some(bdev_flush_complete_cb),
+                cb_arg(s),
+            )
+        };
+        if result < 0 {
+            error!("flush failed for the spdk device, error: {}", result);
+        }
+        match r.await {
```

Comment:
if result was <0 do we still need to await ?

Reply:
handled this scenario. If result < 0, no need to wait, the error should be return at that point.